python/cudf/cudf/core/frame.py

```diff
@@ -2693,7 +2693,13 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype.kind in ("i", "u"):
+            data = self.astype("float64")
```

Comment:
Instead of always type-casting to `float64` we could use min_col_type(self, np.floating) to get a dtype and then we can type-cast self to the respective dtype.

https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/cudf/utils/dtypes.py#L377